Functions/TestResults.Tests.ps1

```diff
@@ -40,6 +40,26 @@ InModuleScope Pester {
                 $xmlTestCase.failure.'stack-trace' | Should -Be 'at line: 28 in  C:\Pester\Result.Tests.ps1'
             }
 
+            It "should write a failed test result when the failure message has an escape character" {
+                #create state
+                $TestResults = New-PesterState -Path TestDrive:\
+                $testResults.EnterTestGroup('Mocked Describe', 'Describe')
+                $time = [TimeSpan]25000000 #2.5 seconds
+                $escape = [string][char]27 # escape, also `e in PowerShell 6+
```

Comment:
@JamesWTruher not 100% sure how to repro the error with escapes on my PowerShell 7, please verify the changes. 

The cast to string here is unnecessary, but I kept it anyway to keep it same as in the code file. (There it is needed to force use of `Replace(string, string)` instead of `Replace(char, char)` which fails.

Reply:
i'll grant you that my test is quite artificial, but here's the way to see it:
``` powershell
PS> Get-Content /tmp/t1.tests.ps1
describe "a simple test" {
    it "can handle an error with an escape sequence" {
       throw  "`e[31m < < , > > `e[0m"
    }
}
```
when run as follows, you can see the error:
``` powershell
[Pester|master↑0↓0•0+0?0] PS> invoke-pester /tmp/t1.tests.ps1 -OutputFile /tmp/xx.xml -OutputFormat NUnitXml
    ____            __
   / __ \___  _____/ /____  _____
  / /_/ / _ \/ ___/ __/ _ \/ ___/
 / ____/  __(__  ) /_/  __/ /
/_/    \___/____/\__/\___/_/
Pester v4.9.0
Executing all tests in '/tmp/t1.tests.ps1'

Executing script /tmp/t1.tests.ps1

  Describing a simple test
    [-] can handle an error with an escape sequence 1ms
      RuntimeException:  < < , > > 
      at <ScriptBlock>, /tmp/t1.tests.ps1: line 3
Tests completed in 24ms
Tests Passed: 0, Failed: 1, Skipped: 0, Pending: 0, Inconclusive: 0 
MethodInvocationException: /Users/jimtru/src/github/forks/JamesWTruher/Pester/Functions/TestResults.ps1
Line |
 591 |             $xmlWriter.WriteElementString('message', $TestResult.FailureMessage)
     |             ^ Exception calling "WriteElementString" with "2" argument(s): "', hexadecimal value 0x1B, is an invalid character."
```

Reply:
I probably should have put that in the test, but it seemed more information than might have been needed.